plugins/modules/azure_rm_appgateway.py

```diff
@@ -2594,6 +2633,25 @@ def get_resource(self):
 
         return False
 
+    def set_web_application_firewall_configuration(self, kwargs):
+        waf_config = dict(kwargs['web_application_firewall_configuration'])
+        if waf_config is None:
+            return
+
+        if 'firewall_policy' in waf_config and waf_config['firewall_policy'] is not None:
+            if 'name' in waf_config['firewall_policy'] and waf_config['firewall_policy']['name'] is not None:
+                waf_config['firewall_policy']['id'] = waf_policy_id(self.subscription_id,
+                                                                    kwargs['resource_group'],
+                                                                    waf_config['firewall_policy']['name'])
+                del waf_config['firewall_policy']['name']
+
+            self.parameters['force_firewall_policy_association'] = waf_config['firewall_policy']['force_association']
+            del waf_config['firewall_policy']['force_association']
+            self.parameters['firewall_policy'] = waf_config['firewall_policy']
+        else:
+            self.module.deprecate("WAF configuration can no longer be added to an application gateway."\
```

Comment:
the backslash is redundant between brackets
```suggestion
            self.module.deprecate("WAF configuration can no longer be added to an application gateway."
```